src/main/java/com/dao/DishOrderDAO.java

```diff
@@ -0,0 +1,188 @@
+package com.dao;
+
+import com.model.DishOrder;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+
+public class DishOrderDAO implements RegularDAO<DishOrder> {
+
+    private DataSource dataSource;
+    private static DishOrderDAO instance;
+
+    private static final String SELECT_BY_ORDER_ID_QUERY = "SELECT * FROM dishes_orders WHERE order_id = ?";
+    private static final String SELECT_BY_ID_QUERY = "SELECT * FROM dishes_orders WHERE id = ?";
+    private static final String SELECT_ALL_QUERY = "SELECT * FROM dishes_orders";
+    private static final String REMOVE_QUERY = "DELETE FROM dishes_orders WHERE id = ?";
+    private static final String UPDATE_QUERY = "UPDATE dishes_orders SET order_id = ?, dish_id = ?, amount = ?, item_sum = ? WHERE id = ?";
+    private static final String INSERT_QUERY = "INSERT INTO dishes_orders VALUES (NULL, ?, ?, ?, ?)";
+
+    private DishOrderDAO() {
+
+    }
+
+    public static DishOrderDAO getInstance() {
+        if (instance == null)
+            instance = new DishOrderDAO();
+
+        return instance;
+    }
+
+
+    @Override
+    public void setDataSource(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public Optional<DishOrder> getById(long id) {
+
+        DishOrder dishOrder = null;
+
+        try (Connection con = dataSource.getConnection()) {
+
+            PreparedStatement sql = con.prepareStatement(SELECT_BY_ID_QUERY);
+            sql.setLong(1, id);
+            ResultSet rs = sql.executeQuery();
+
+            if (rs.next())
+            {
+                dishOrder = createDishOrderEntity(rs);
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return Optional.ofNullable(dishOrder);
+    }
+
+    @Override
+    public List<DishOrder> getAll() {
+
+        List<DishOrder> users = new ArrayList<>();
+
+        try (Connection connection = dataSource.getConnection()) {
+
+            PreparedStatement sql = connection.prepareStatement(SELECT_ALL_QUERY);
+
+            ResultSet rs = sql.executeQuery();
+
+            while (rs.next())
+                users.add(createDishOrderEntity(rs));
+
+        } catch (SQLException e) {
+
+            e.printStackTrace();
+        }
+
+        return users;
```

Comment:
why this variable is called "users"?
you should rename according to its content